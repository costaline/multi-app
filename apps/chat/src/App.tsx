import { Route, Routes } from '@solidjs/router'
import { lazy } from 'solid-js'

import type { Component } from 'solid-js'

const Main = lazy(async () => await import('./pages'))
const Room = lazy(async () => await import('./pages/room'))
const NotFound404 = lazy(async () => await import('./pages/404'))

const App: Component = () => {
	return (
		<Routes>
			<Route path="/room/:id" component={Room} />
			<Route path="/" component={Main} />
			<Route path="/*" component={NotFound404} />
		</Routes>
	)
}

export default App
