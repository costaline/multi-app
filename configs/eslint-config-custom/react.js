module.exports = {
	extends: [
		'./index.js',
		'plugin:react/recommended',
		'plugin:react/jsx-runtime',
		'plugin:react-hooks/recommended',
	],

	settings: {
		react: {
			version: 'detect',
		},
	},

	rules: {
		'react/button-has-type': 'warn',
		'react/destructuring-assignment': [
			'warn',
			'always',
			{
				ignoreClassFields: true,
			},
		],
		'react/function-component-definition': 'off',
		'react/jsx-curly-brace-presence': ['warn', 'never'],
		'react/jsx-filename-extension': [
			'warn',
			{ extensions: ['js', 'jsx', '.tsx'] },
		],
		'react/jsx-fragments': 'off',
		'react/jsx-key': ['warn', { checkKeyMustBeforeSpread: true }],
		'react/jsx-no-useless-fragment': ['warn', { allowExpressions: true }],
		'react/jsx-props-no-spreading': 'off',
		'react/jsx-sort-props': [
			'warn',
			{
				callbacksLast: true,
				shorthandFirst: true,
				shorthandLast: false,
				ignoreCase: false,
				noSortAlphabetically: false,
				reservedFirst: true,
			},
		],
		'react/jsx-uses-vars': 'warn',
		'react/self-closing-comp': ['warn', { component: true, html: true }],

		'react/forbid-prop-types': ['warn', { forbid: ['any', 'array'] }],
		'react/no-unused-prop-types': 'warn',
		'react/prop-types': 'warn',
		'react/require-default-props': 'off',
	},

	overrides: [
		{
			files: ['*.ts', '*.tsx'],

			rules: {
				'@typescript-eslint/no-restricted-imports': [
					'warn',
					{
						name: 'react-redux',
						importNames: ['useSelector', 'useDispatch'],
						message:
							'Use typed hooks `useAppDispatch` and `useAppSelector` instead.',
					},
				],
			},
		},
	],
}
