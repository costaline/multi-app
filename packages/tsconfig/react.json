{
	"$schema": "https://json.schemastore.org/tsconfig",
	"display": "React",
	"extends": "./base.json",
	"compilerOptions": {
		"useDefineForClassFields": true,
		"lib": ["DOM", "DOM.Iterable", "ESNext"],
		"noEmit": true,
		"jsx": "react-jsx"
	}
}
