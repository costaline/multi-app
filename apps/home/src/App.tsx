import { useState } from 'react'
// import {MyElement} from 'wc'

import './App.css'
import reactLogo from './assets/react.svg'

function App() {
	const [count, setCount] = useState(0)



	return (
		<div className="App">
			<button>click</button>
			<div>
				<a href="https://vitejs.dev" rel="noreferrer" target="_blank">
					{' '}
					<img alt="Vite logo" className="logo" src="/vite.svg" />
				</a>
				<a href="https://reactjs.org" rel="noreferrer" target="_blank">
					<img alt="React logo" className="logo react" src={reactLogo} />
				</a>
			</div>

			{/*<my-element docsHint={`hello from counter: ${count}`}></my-element>*/}
			<demo-element name="REACT">Some text</demo-element>

			<global-nav/>

			<h1>Vite + React</h1>
			<div className="card">
				<button onClick={() => setCount((count) => count + 1)}>
					count is {count}
				</button>
				<p>
					Edit <code>src/App.tsx</code> and save to test HMR
				</p>
			</div>
			<p className="read-the-docs">
				Click on the Vite and React logos to learn more
			</p>
		</div>
	)
}

export default App
