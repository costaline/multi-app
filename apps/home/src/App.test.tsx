import { fireEvent, render, screen } from '@testing-library/react'

import App from './App'

describe('Test app component', () => {
	test('The component has the main title', () => {
		render(<App />)

		expect(screen.getByRole('heading', { level: 1 })).toBeTruthy()
	})

	test('Counter works', () => {
		render(<App />)

		const $btn = screen.getByText(/count is/i)

		expect($btn.textContent).toContain(0)
		fireEvent.click($btn)
		expect($btn.textContent).toContain(1)
	})

	test('Snapshot', () => {
		const view = render(<App />)

		expect(view).toMatchSnapshot()
	})
})
