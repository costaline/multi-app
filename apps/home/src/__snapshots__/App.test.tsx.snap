// Vitest Snapshot v1

exports[`Test app component > Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <button>
          click
        </button>
        <div>
          <a
            href="https://vitejs.dev"
            rel="noreferrer"
            target="_blank"
          >
             
            <img
              alt="Vite logo"
              class="logo"
              src="/vite.svg"
            />
          </a>
          <a
            href="https://reactjs.org"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="React logo"
              class="logo react"
              src="/src/assets/react.svg"
            />
          </a>
        </div>
        <demo-element
          name="REACT"
        >
          Some text
        </demo-element>
        <global-nav />
        <h1>
          Vite + React
        </h1>
        <div
          class="card"
        >
          <button>
            count is 
            0
          </button>
          <p>
            Edit 
            <code>
              src/App.tsx
            </code>
             and save to test HMR
          </p>
        </div>
        <p
          class="read-the-docs"
        >
          Click on the Vite and React logos to learn more
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <button>
        click
      </button>
      <div>
        <a
          href="https://vitejs.dev"
          rel="noreferrer"
          target="_blank"
        >
           
          <img
            alt="Vite logo"
            class="logo"
            src="/vite.svg"
          />
        </a>
        <a
          href="https://reactjs.org"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="React logo"
            class="logo react"
            src="/src/assets/react.svg"
          />
        </a>
      </div>
      <demo-element
        name="REACT"
      >
        Some text
      </demo-element>
      <global-nav />
      <h1>
        Vite + React
      </h1>
      <div
        class="card"
      >
        <button>
          count is 
          0
        </button>
        <p>
          Edit 
          <code>
            src/App.tsx
          </code>
           and save to test HMR
        </p>
      </div>
      <p
        class="read-the-docs"
      >
        Click on the Vite and React logos to learn more
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
